Update guide, readme and fix issue in colormapping

wi-re opened the pull request:
Update readme to include reference to guide

Add guide for mouse interaction, colormapping and realtime timestepping
Remove GLFW key interaction as it is not necessary and too convoluted

Fix bug in colormapping where the file was not being read correctly as it had hardcoded windows file endings ('\r\n'), which made it impossible to pick colormaps on linux. This is fixed by stripping the readline of the '\r' character and '\n' characters.

wi-re commented:
I've now also changed the guide to use z direction for gravity to fit the physics that are expected more closely. This should also address #9

Ceyron commented:
Could you disentangle your contributions into separate PRs:

1. Fix of camera positions
2. Removal of GLFW section
3. Introduction to capturing the mouse position
4. Improval to Colormaps and introduction
5. Linking to the Guide in the README

That should simplify reviewing and keeping accounts. For (1.), I'd suggest we re-open #9.

wi-re commented:
Re-opened for now. This will be split before merging to main by separating out the colormaps fix as this does not really belong in the guide changes and the guide changes work irrespective of the fix.

Ceyron commented:
Thanks 👍 
I agree, the changes to the guide can be made jointly.

Ceyron commented:
Thanks for all the work :+1: 
I removed the parts related to Colormapping (now handled by #11).

Ceyron commented on guide.md:
```diff
@@ -531,138 +533,87 @@ Which are all necessary changes. You may still want to try the GLFW based varian
+        continue;
+    auto dx = drag.x * right;
+    auto dy = -drag.y * up;
+    object.force += dx + dy;
```
1. I think it would be helpful if we added the hint that this section of code should be in the `Scene1::simulateStep()` method (similar to where we would have the button interaction as well)
2. I can imagine it would also be helpful if it showed how the force could act on the particles. Something simple would be
```cpp
if(ImGui::IsMouseReleased(ImGuiMouseButton_Right)){   
    auto drag = ImGui::GetMouseDragDelta(1);
    if(drag.x == 0 && drag.y == 0)
        continue;
    auto dx = drag.x * right;
    auto dy = -drag.y * up;
    auto force += dx + dy;
    for (auto& particle : particles){
        particle.position += force * 0.01f;
     }
}
```
This also shows that the force must be further discounted with the time step size.

Ceyron reviewed (commented):
Hi @wi-re,

I added a remark on the mouse input. One more general comment:

Since the GLFW section is now removed, I think some other lines require modification, specifically:
- [ ] Line 476
- [ ] Line 488
- [ ] Line 532

wi-re commented:
> Hi @wi-re,
> 
> I added a remark on the mouse input. One more general comment:
> 
> Since the GLFW section is now removed, I think some other lines require modification, specifically:
> 
> * [ ]  Line 476
> * [ ]  Line 488
> * [ ]  Line 532

476 + 488: Line 476 we can keep, it is still worth pointing out that there are alternative ways to approach this problem and if someone wants to play around with it, I think it makes sense to point to the references. Similarly 488 is still important if we keep the pointer to GLFW input as doing that in the normal tutorial submissions would lead to issues and I can imagine a scenario where a group might want to use a different style of input handling for their open project

532: A more likely candidate to deletion and I would probably remove the comment about glfw

Ceyron commented:
> 476 + 488: Line 476 we can keep, it is still worth pointing out that there are alternative ways to approach this problem and if someone wants to play around with it, I think it makes sense to point to the references. Similarly 488 is still important if we keep the pointer to GLFW input as doing that in the normal tutorial submissions would lead to issues and I can imagine a scenario where a group might want to use a different style of input handling for their open project

I am inclined to leave out the GLFW parts altogether to keep the guide as concise as possible. Using an imgui-based input is sufficient to solve all programming exercises, isn't it? If you want to keep them, then they require a slight re-wording because, for example:

- `for which we will use an immediate mode input via ImGUI (recommended). An alternative is to use an event based input mode via GLFW key inputs ([more information here](https://www.glfw.org/docs/3.3/input_guide.html#input_key)).`
- ...

> 532: A more likely candidate to deletion and I would probably remove the comment about glfw

Great :+1:.

Ceyron reviewed (approved):
LGTM,

I would suggest we merge it on Wednesday since the deadline for the zeroth submission is Tuesday, 23:59. What do you think?

wi-re commented:
Let's do that, that should still give people plenty of time to include the mouse interaction based on this guide and won't *take away* anything they've relied upon so far (albeit the GLFW variant is unlikely to be used by anyone at this point).
